tests/test_interactive.py

```
@@ -1853,3 +1853,94 @@ def test_returns_none_when_cancelled(self):
+        self.assertEqual(out, "red")
+
+    def test_filesystem_spill_marker_at_boundary(self):
+        """Marker is inserted at the exact char boundary reported by spill."""
+        from holmes.interactive import _build_show_output
+        full = "A" * 40 + "B" * 60
+        boundary = 40
+        result = self._result(
+            data="llm-facing preview (boilerplate + slice)",
+            original_stringified_data=full,
+            llm_preview_boundary_chars=boundary,
+            spilled_file_path="/tmp/holmes/tool-xyz.json",
```

Ruff (0.15.10)

[error] Probable insecure usage of temporary file or directory: "/tmp/holmes/tool-xyz.json"

(S108)